docs/cookbook/named_entity_recognition.md

```diff
@@ -0,0 +1,422 @@
+# Named Entity Recognition
+
+In this guide, we demonstrate techniques to perform Named Entity Recognition (NER) using Large Language Models (LLMs). We will use Groq's llama-3.1-8b-instant model, but you can adapt this approach to other models with similar capabilities.
+
+??? tip "Mirascope Concepts Used"
+
+    - [Calls](../learn/calls.md)
+    - [Response Model](../learn/response_models.md)
+    - [Prompt Templates](../learn/prompt_templates.md)
+
+!!! note "Background"
+
+    Named Entity Recognition is a subtask of information extraction that seeks to locate and classify named entities in text into predefined categories such as person names, organizations, locations, etc. LLMs have revolutionized NER by enabling more context-aware and hierarchical entity recognition, going beyond traditional rule-based or statistical methods.
+
+## Setup and Imports
+
+First, let's set up our environment with the necessary imports:
+
+```python
+from mirascope.core import groq, prompt_template
+from pydantic import BaseModel, Field
+from typing import List, Optional
+```
+
+## Defining Data Models
+
+We'll use Pydantic models to structure our entity data:
+
+```python
+class NamedEntity(BaseModel):
+    entity: str = Field(description="The entity found in the text")
+    label: str = Field(description="The label of the entity (e.g., PERSON, ORGANIZATION, LOCATION)")
+    parent: Optional[str] = Field(description="The parent entity if this entity is nested within another entity", default=None)
+    children: List['NamedEntity'] = Field(default_factory=list, description="Nested entities within this entity")
+
+class NamedEntityRecognition(BaseModel):
+    entities: List[NamedEntity] = Field(description="List of top-level named entities found in the text")
+    given_text: str = Field(description="The text that was checked for named entities")
+```
+
+These models allow us to capture hierarchical relationships between entities, which is a key feature of this advanced NER approach.
+
+## Implementing the NER Function
+
+Now, let's create our main NER function using Groq's LLM:
+
+```python
+@groq.groq_call(
```

Comment:
nit: just use `groq.call` here for enhanced readability

Reply:
Will be fixing it 👍🏼

Reply:
Fixed in new commit